# (PART) BIOINFORMATICS ANALYSIS {-}

# Preprocessing 16S rRNA Sequencing Data{#preprocess-16S-reads}
At this point we are aware of the preprocessing and bioinformatics tools needed to getting started with microbiome data analysis. We assume that the tools are already installed. Now it is time to use them to:

- Get simple statistics.
- Check the quality of the reads.
- Create summary report of quality metrics. 
- Trim poor read at a user-specified cutoff.
- Remove contaminants.


## Original read statistics
Tool: SeqKit.

```bash
mkdir -p data
mkdir -p data/stats1  
seqkit stat *.fastq.gz >data/stats1/seqkit_stats.txt
```

## Initial read quality scores
Tool: FastQC

```bash
mkdir data/fastqc1
fastqc *.fastq.gz -o data/fastqc1
```

## Summary of initial read quality scores
Tool: MultQC.
```bash
mkdir data/multiqc1
multiqc -f --data-dir data/fastqc1 -o data/multiqc1 --export
```

> Plots from `multiqc` are exported to `multiqc_plots` folder. 

![Original: Mean quality scores](data/multiqc1/multiqc_plots/png/mqc_fastqc_per_base_sequence_quality_plot_1.png){width=100%}

<br>

![Original: Per sequence quality scores](data/multiqc1/multiqc_plots/png/mqc_fastqc_per_sequence_quality_scores_plot_1.png){width=100%}

<br>

## Trimming poor reads 
- Using `bbduk.sh` from bbmap platform
- We use `-Xmx4g` switch to tell Java how much memory (heap size) to use, in this demo is 4 GB.
- Note that we will use a `for loop` to specify the file name pattern to look for. Some files may contain different patterns containing `R1_001.fastq.gz`. Here we use files downloaded from NCBI-SRA and looks like `SRR7450758_1.fastq.gz`.
- Then we will rerun `seqkit stat`, `fastqc` and `multiqc` on the trimmed reads.

```bash
for i in `ls -1 *_1.fastq.gz | sed 's/_1.fastq.gz//'`
  do
  bbduk.sh -Xmx4g in1=$i\_1.fastq.gz in2=$i\_2.fastq.gz out1=data/trimmed/$i\_1.fastq.gz out2=data/trimmed/$i\_2.fastq.gz qtrim=r trimq=25 overwrite=True
  done
```

```bash
mkdir -p data/stats2  
seqkit stat data/trimmed/*.fastq.gz >data/stats2/seqkit_stats.txt

mkdir data/fastqc2
fastqc data/trimmed/*.fastq.gz -o data/fastqc2

mkdir data/multiqc2
multiqc -f --data-dir data/fastqc2 -o data/multiqc2 --export
```

![Trimmed: Mean quality scores](data/multiqc2/multiqc_plots/png/mqc_fastqc_per_base_sequence_quality_plot_1.png){width=100%}

<br>

![Trimmed: Per sequence quality scores](data/multiqc2/multiqc_plots/png/mqc_fastqc_per_sequence_quality_scores_plot_1.png){width=100%}

## Read decontamination
- Using `bbduk.sh` on trimmed reads.
- This will remove some contamination (if found), e.g. phiX Control reads.
- Then we will rerun `seqkit stat`, `fastqc` and `multiqc` on the decontaminated reads.

```bash
for i in `ls -1 *_1.fastq.gz | sed 's/_1.fastq.gz//'`
do
bbduk.sh -Xmx4g in1=data/trimmed/$i\_1.fastq.gz in2=data/trimmed/$i\_2.fastq.gz out1=data/decontam/$i\_1.fastq.gz out2=data/decontam/$i\_2.fastq.gz outm1=data/decontam/matchedphix/$i\_1.fastq.gz outm2=data/decontam/matchedphix/$i\_2.fastq.gz ref=~/bbmap/resources/phix174_ill.ref.fa.gz k=31 hdist=1 overwrite=True
done
```

```bash
mkdir -p data/stats3  
seqkit stat data/decontam/*.fastq.gz >data/stats3/seqkit_stats.txt

mkdir data/fastqc3
fastqc data/decontam/*.fastq.gz -o data/fastqc3

mkdir data/multiqc3
multiqc -f --data-dir data/fastqc3 -o data/multiqc3 --export
```

<br>

![Decontaminated: Mean quality scores](data/multiqc3/multiqc_plots/png/mqc_fastqc_per_base_sequence_quality_plot_1.png){width=100%}

<br>

![Decontaminated: Per sequence quality scores](data/multiqc3/multiqc_plots/png/mqc_fastqc_per_sequence_quality_scores_plot_1.png){width=100%}

<br>

## Merged and compare preprocessed reads
```{r message=FALSE, warning=FALSE}
library(tidyverse, suppressPackageStartupMessages())
library(ggtext)

stats1 <- read_table("data/stats1/seqkit_stats.txt", show_col_types = F) %>% 
  mutate(file = str_replace_all(file, ".*/", "")) %>% 
  select(file, original = num_seqs)

stats2 <- read_table("data/stats2/seqkit_stats.txt", show_col_types = F) %>% 
  mutate(file = str_replace_all(file, ".*/", "")) %>% 
  select(file, trimmed = num_seqs)

stats3 <- read_table("data/stats3/seqkit_stats.txt", show_col_types = F) %>% 
  mutate(file = str_replace_all(file, ".*/", "")) %>% 
  select(file, decontaminated = num_seqs)

inner_join(stats1, stats2, by = "file") %>% 
  inner_join(., stats3, by = "file") %>% 
  mutate(strand = ifelse(str_detect(file, "_1"), "foward", "reverse"), .before=original) %>%
  pivot_longer(cols = -c(file, strand), names_to = "variable", values_to = "num_seqs") %>% 
  mutate(variable = factor(variable),
         variable = fct_reorder(variable, num_seqs, .desc=TRUE)) %>% 
  ggplot(aes(x = strand, y = num_seqs/1000, fill = variable)) +
  geom_col(position = "dodge") +
  labs(x = "Read Strand", y = "Number of Reads (thousand)", fill = "Preprocess") +
  theme_classic() +
  theme(axis.text.x = element_markdown(),
        legend.text = element_text(face = NULL),
        legend.key.size = unit(12, "pt")) + 
  scale_y_continuous(expand = c(0, 0))
```


<br>

# Preprocessing Metagenomics Sequencing Data {#preprocess-kneaddata}

## Getting initial read statistics 
```bash
mkdir -p kneaddata
mkdir -p kneaddata/stats1  
seqkit stat *.fastq.gz >kneaddata/stats1/seqkit_stats.txt
```

## Initial read quality scores
- Using KneadData tool from [biobakery](https://github.com/biobakery/kneaddata).
- The `kneaddata` tool is exclusively used to preprocess the metagenomics data.
- We will set it to run `fastqc` to compute initial quality checks.
- Then summarize the `fastqc` results outside KneadData pipeline using `multiqc` software.

```bash
for i in data/*.fastq.gz
do
time kneaddata --i $i \
-o kneaddata/fastqc1 \
--run-fastqc-start \
--bypass-trim \
--bypass-trf \
--sequencer-source "none" \
-t 4 
done
```

```bash
mkdir -p kneaddata/multiqc1
multiqc -f --data-dir kneaddata/fastqc1 -o kneaddata/multiqc1 --export
```

## Read Trimming
- Using `trimmomatic` tool to trim the poor reads.
- The `fastqc` function is automatically run after trimming.
- Then we will run `seqkit` and `multiqc` on the trimmed reads

```bash
for i in data/*.fastq.gz
do
  time kneaddata --i $i \
    -o kneaddata/fastqc2 \
    --trimmomatic /Users/tmbuza/opt/anaconda3/envs/biobakery3/bin/ \
    --trimmomatic-options \
        "ILLUMINACLIP:/Volumes/SeagateTMB/trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 \
        LEADING:3 \
        TRAILING:3 \
        SLIDINGWINDOW:4:20 \
        MINLEN:60" \
    --sequencer-source "NexteraPE" \
    --run-fastqc-end \
    --bypass-trf \
    -t 4 
done
```

```bash
mkdir -p kneaddata/stats2
seqkit stat kneaddata/fastqc2/*trimmed.fastq >kneaddata/stats2/seqkit_stats.txt

mkdir -p kneaddata/multiqc2
multiqc -f --data-dir kneaddata/fastqc2 -o kneaddata/multiqc2 --export
```


## Read Decontamination
### Option 1: Searching against Bowtie2 database
- Trimming is done first using `trimmomatic`.
- Then contaminated reads are identified by searching the reads against a `bowtie2` reference database.
- The `fastqc` is automatically run after decontamination.
- Then we will run `seqkit` and `multiqc` on the decontaminated reads

```bash
for i in data/*.fastq.gz
do
time kneaddata --i $i \
  -o kneaddata/fastqc3 \
      --reference-db /Volumes/SeagateTMB/kneaddata_database/ \
      --trimmomatic /Users/tmbuza/opt/anaconda3/envs/biobakery3/bin/ \
      --trimmomatic-options \
          "ILLUMINACLIP:/Volumes/SeagateTMB/trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 \
          LEADING:3 \
          TRAILING:3 \
          SLIDINGWINDOW:4:20 \
          MINLEN:60" \
      --sequencer-source "NexteraPE" \
      --run-trf \
      --run-fastqc-end \
      -t 4 
done
```

```bash
mkdir -p kneaddata/stats3
seqkit stat kneaddata/fastqc3/*trimmed.fastq >kneaddata/stats3/seqkit_stats.txt

mkdir -p kneaddata/multiqc3
multiqc -f --data-dir kneaddata/fastqc3 -o kneaddata/multiqc3 --export
```

### Option 2: Seaching against BMTagger database 
- Trimming is done first using `trimmomatic`.
- Then contaminated reads are identified by searching the reads against `BMTagger` database.
- The `fastqc` is automatically run after decontamination.
- Then we will run `seqkit` and `multiqc` on the decontaminated reads

```bash
mkdir kneaddata
for i in data/*.fastq.gz
do
time kneaddata --i $i \
  -o kneaddata/fastqc4 \
      --reference-db /Volumes/SeagateTMB/Human_Assembly19_BMTagger_DB/ \
      --trimmomatic /Users/tmbuza/opt/anaconda3/envs/biobakery3/bin/ \
      --trimmomatic-options \
          "ILLUMINACLIP:/Volumes/SeagateTMB/trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 \
          LEADING:3 \
          TRAILING:3 \
          SLIDINGWINDOW:4:20 \
          MINLEN:60" \
      --sequencer-source "NexteraPE" \
      --run-trf \
      --run-bmtagger \
      --run-fastqc-end \
      -t 4 
done
```

```bash
mkdir -p kneaddata/stats4
seqkit stat kneaddata/fastqc4/*trimmed.fastq >kneaddata/stats4/seqkit_stats.txt

mkdir -p kneaddata/multiqc4
multiqc -f --data-dir kneaddata/fastqc4 -o kneaddata/multiqc4 --export
```


