# Taxonomic Profiling Using MetaPhlAn Pipeline {#metaphlan-pipeline}

## Generating relative abundance {#rel-abund-noread_stats}
- The `metaphlan` loop through the fastq files using specified parameters.
- Then the files are merged into an abundance table using the `merge_metaphlan_tables.py` python script and store the output in a folder named *metaphlan3_profiles*.

> Note: The output does not contain read count. See next section on how to add mapped read count.

```bash
for i in data/*.fastq.gz
	do 
		metaphlan $i \
		--input_type fastq \
		--force \
		--bowtie2db /Volumes/SeagateTMB/metaphlan_databases/ \
		--nproc 4 \
		--stat_q 0.02 \
		--min_mapq_val 5 \
		--output_file ${i%}_metaphlan3_profile.txt \
		--bowtie2out ${i%}_metaphlan3_bowtie2out.txt
	done
	
mkdir -p metaphlan3_profiles
merge_metaphlan_tables.py metaphlan3_profiles/*_metaphlan3_profile*.txt > metaphlan3_profiles/merged_abundance_table.txt

```

## Generating relative abundance with read statistics {#rel-abund-read_stats}
- We add `-t rel_ab_w_read_stats` which will add the read statistics to the output.
- We also add an option of generating a `biom` file which can be useful in some analyses.

> Merging the output using the `merge_metaphlan_tables.py` python script will ignore the read statistics. In the next section we will demonstrate how to process the abundance files to include read statistics in the merged file.


```bash
for i in data/*.fastq.gz
	do 
		metaphlan $i \
		--input_type fastq \
		--force \
		--bowtie2db /Volumes/SeagateTMB/metaphlan_databases/ \
		--nproc 4 \
		--stat_q 0.02 \
		--min_mapq_val 5 \
		--output_file ${i%}_metaphlan3_profile.txt \
		--bowtie2out ${i%}_metaphlan3_bowtie2out.txt \
		--biom ${i%}_metaphlan3_abundance.biom \
    -t rel_ab_w_read_stats
	done
```

## Merging `metaphlan` abundance files
- First we create an object to specify the location of `metaphlan` output.
- Then we an empty data frame containing an empty column for holding the file names.
- Then we loop through each file to extract the output and create a merged abundance table.
- We will then create a tidy table containing sample ID, taxonomy, relative abundance and read statistics.

> Note: We will specify the path where the `metaphlan` output is located. Replace the path to match yours.

```{r}
path <- "../smda-end2end/data/metagenomics/metaphlan3_profiles/*_profile.txt"
files <- Sys.glob(path)

merged_files <- data.frame(
  ModelName = character(), 
  Object = character(), 
  stringsAsFactors = FALSE) %>%
  mutate(Filename = "")

for (i in 1:length(files)){
  currentFile = read_delim(files[i], skip = 4, show_col_types = FALSE) %>% 
    mutate(Filename = files[i])
    merged_files = rbind(merged_files, currentFile)}

m3_merged_abund_table <- merged_files %>% 
  rename_all(tolower) %>%
  rename(clade_name = colnames(merged_files)[1],
         sample_id = filename,
         rel_abund = relative_abundance, 
         mapped_read = estimated_number_of_reads_from_the_clade) %>% 
  mutate(sample_id = str_replace_all(sample_id, ".*/", ""),
         sample_id = str_replace_all(sample_id, ".fast.*", "")) %>%
  select(sample_id, taxonomy = clade_name, rel_abund, mapped_read)
    
colnames(m3_merged_abund_table)

saveRDS(m3_merged_abund_table, "RDataRDS/m3_merged_abund_table.rds")
resave(m3_merged_abund_table, file = "RDataRDS/final-bioinfo_objects.RData")
```



