<!-- # (PART) METAGENOMIC PROFILING {-} -->

# Taxonomic Profiling Using MetaPhlAn Pipeline {#metaphlan-pipeline}

## Generating relative abundance
- The `metaphlan` loop through the fastq files using specified parameters.
- Then the files are merged into an abundance table using the `merge_metaphlan_tables.py` python script and store the output in a folder named *metaphlan3_profiles*.

> Note: The output does not contain read count. See next section on how to add mapped read count.

```bash
for i in data/*.fastq.gz
	do 
		metaphlan $i \
		--input_type fastq \
		--force \
		--bowtie2db /Volumes/SeagateTMB/metaphlan_databases/ \
		--nproc 4 \
		--stat_q 0.02 \
		--min_mapq_val 5 \
		--output_file ${i%}_metaphlan3_profile.txt \
		--bowtie2out ${i%}_metaphlan3_bowtie2out.txt
	done
	
mkdir -p metaphlan3_profiles
merge_metaphlan_tables.py metaphlan3_profiles/*_metaphlan3_profile*.txt > metaphlan3_profiles/merged_abundance_table.txt

```

## Generating relative abundance and read statistics
- We add `-t rel_ab_w_read_stats` which will add the read statistics to the output.
- We also add an option of generating a `biom` file which can be useful in some analyses.

> Merging the output using the `merge_metaphlan_tables.py` python script will ignore the read statistics. In the next section we will demonstrate how to process the abundance files to include read statistics in the merged file.


```bash
for i in data/*.fastq.gz
	do 
		metaphlan $i \
		--input_type fastq \
		--force \
		--bowtie2db /Volumes/SeagateTMB/metaphlan_databases/ \
		--nproc 4 \
		--stat_q 0.02 \
		--min_mapq_val 5 \
		--output_file ${i%}_metaphlan3_profile.txt \
		--bowtie2out ${i%}_metaphlan3_bowtie2out.txt \
		--biom ${i%}_metaphlan3_abundance.biom \
    -t rel_ab_w_read_stats
	done
```

