<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 3 Microbial Profiling | Bioinformatics Analysis of Microbiome Data</title>
<meta name="author" content="Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA">
<meta name="description" content="3.1 Generalized bioinformatics workflow    3.2 Format of input data The default format used in this practical user guide is compressed fastq, *.fastq.gz.  Compressing fastq files using gzip  Run...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Chapter 3 Microbial Profiling | Bioinformatics Analysis of Microbiome Data">
<meta property="og:type" content="book">
<meta property="og:description" content="3.1 Generalized bioinformatics workflow    3.2 Format of input data The default format used in this practical user guide is compressed fastq, *.fastq.gz.  Compressing fastq files using gzip  Run...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 3 Microbial Profiling | Bioinformatics Analysis of Microbiome Data">
<meta name="twitter:description" content="3.1 Generalized bioinformatics workflow    3.2 Format of input data The default format used in this practical user guide is compressed fastq, *.fastq.gz.  Compressing fastq files using gzip  Run...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/_Sen-0.4.1/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/_Bitter-0.4.1/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides">Bioinformatics Analysis of Microbiome Data</a>:
        <small class="text-muted">End-to-End Practical User Guides</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Bioinformatics Analysis of Microbiome Data</a></li>
<li class="book-part">BIOINFORMATICS ANALYSIS</li>
<li><a class="" href="preprocess-16S-reads.html"><span class="header-section-number">1</span> Standard Sequence Preprocessing</a></li>
<li><a class="" href="preprocess-kneaddata.html"><span class="header-section-number">2</span> Preprocessing Metagenomics Data Using Kneaddata</a></li>
<li><a class="active" href="microbial-profiling.html"><span class="header-section-number">3</span> Microbial Profiling</a></li>
<li><a class="" href="mothur-taxonomic-profiling.html"><span class="header-section-number">4</span> Mcrobial profiling Using mothur Pipeline</a></li>
<li><a class="" href="qiime2-taxonomic-profiling.html"><span class="header-section-number">5</span> Mcrobial Profiling Using qiime2 Pipeline</a></li>
<li><a class="" href="metaphlan-taxonomic-profiling.html"><span class="header-section-number">6</span> Taxonomic Profiling of Metagenomics Sequencing Data</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></li>
<li><a class="" href="software-and-packages.html"><span class="header-section-number">B</span> Software and Packages</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/CDI2021/smda_part2">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="microbial-profiling" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> Microbial Profiling<a class="anchor" aria-label="anchor" href="#microbial-profiling"><i class="fas fa-link"></i></a>
</h1>
<div id="generalized-bioinformatics-workflow" class="section level2" number="3.1">
<h2>
<span class="header-section-number">3.1</span> Generalized bioinformatics workflow<a class="anchor" aria-label="anchor" href="#generalized-bioinformatics-workflow"><i class="fas fa-link"></i></a>
</h2>
<p><a href=""><img src="img/part2_flow.png" alt="Book cover" width="100%" style="padding: 20px 10px 30px 10px; float: right;"></a></p>
<p><br></p>
</div>
<div id="format-of-input-data" class="section level2" number="3.2">
<h2>
<span class="header-section-number">3.2</span> Format of input data<a class="anchor" aria-label="anchor" href="#format-of-input-data"><i class="fas fa-link"></i></a>
</h2>
<p>The default format used in this practical user guide is compressed fastq, <code>*.fastq.gz</code>.</p>
<div class="tmbalert">
<h2 id="compressing-fastq-files-using-gzip">
Compressing fastq files using <code>gzip</code>
</h2>
<blockquote>
<p>
Run the command below only if the sequencing data files are not compressed!
</p>
</blockquote>
<p>
gzip *.fastq
</p>
</div>
<p><br></p>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="preprocess-kneaddata.html"><span class="header-section-number">2</span> Preprocessing Metagenomics Data Using Kneaddata</a></div>
<div class="next"><a href="mothur-taxonomic-profiling.html"><span class="header-section-number">4</span> Mcrobial profiling Using mothur Pipeline</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#microbial-profiling"><span class="header-section-number">3</span> Microbial Profiling</a></li>
<li><a class="nav-link" href="#generalized-bioinformatics-workflow"><span class="header-section-number">3.1</span> Generalized bioinformatics workflow</a></li>
<li><a class="nav-link" href="#format-of-input-data"><span class="header-section-number">3.2</span> Format of input data</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/CDI2021/smda_part2/blob/main/02b_bioinfomatics_analysis.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/CDI2021/smda_part2/edit/main/02b_bioinfomatics_analysis.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p><strong>Bioinformatics Analysis of Microbiome Data</strong> is PART 2 of the <a class="text-light" href="https://complexdatainsights.com/books/microbiome-analysis-in-R/"><u>End-to-End Practical User Guides</u></a> written by Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA. 
  </p>
</div>

  <div class="col-12 col-md-6 mt-3">
    <p>This guide was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div>
</div>
</footer>
</body>
</html>
