<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Mcrobial profiling Using mothur Pipeline | Bioinformatics Analysis of Microbiome Data</title>
<meta name="author" content="Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA">
<meta name="description" content="4.1 Download a Mothur trained classifer For demo purposes we will use Silva seed due to its smaller size. Other classifiers can be found here. wget...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Chapter 4 Mcrobial profiling Using mothur Pipeline | Bioinformatics Analysis of Microbiome Data">
<meta property="og:type" content="book">
<meta property="og:description" content="4.1 Download a Mothur trained classifer For demo purposes we will use Silva seed due to its smaller size. Other classifiers can be found here. wget...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Mcrobial profiling Using mothur Pipeline | Bioinformatics Analysis of Microbiome Data">
<meta name="twitter:description" content="4.1 Download a Mothur trained classifer For demo purposes we will use Silva seed due to its smaller size. Other classifiers can be found here. wget...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/_Sen-0.4.1/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/_Bitter-0.4.1/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides">Bioinformatics Analysis of Microbiome Data</a>:
        <small class="text-muted">End-to-End Practical User Guides</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Bioinformatics Analysis of Microbiome Data</a></li>
<li class="book-part">BIOINFORMATICS ANALYSIS</li>
<li><a class="" href="preprocess-16S-reads.html"><span class="header-section-number">1</span> Standard Sequence Preprocessing</a></li>
<li><a class="" href="preprocess-kneaddata.html"><span class="header-section-number">2</span> Preprocessing Metagenomics Data Using Kneaddata</a></li>
<li><a class="" href="microbial-profiling.html"><span class="header-section-number">3</span> Microbial Profiling</a></li>
<li><a class="active" href="mothur-taxonomic-profiling.html"><span class="header-section-number">4</span> Mcrobial profiling Using mothur Pipeline</a></li>
<li><a class="" href="qiime2-taxonomic-profiling.html"><span class="header-section-number">5</span> Mcrobial Profiling Using qiime2 Pipeline</a></li>
<li><a class="" href="metaphlan-taxonomic-profiling.html"><span class="header-section-number">6</span> Taxonomic Profiling of Metagenomics Sequencing Data</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></li>
<li><a class="" href="software-and-packages.html"><span class="header-section-number">B</span> Software and Packages</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/CDI2021/smda_part2">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="mothur-taxonomic-profiling" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Mcrobial profiling Using <code>mothur</code> Pipeline<a class="anchor" aria-label="anchor" href="#mothur-taxonomic-profiling"><i class="fas fa-link"></i></a>
</h1>
<div id="download-a-mothur-trained-classifer" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Download a Mothur trained classifer<a class="anchor" aria-label="anchor" href="#download-a-mothur-trained-classifer"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>For demo purposes we will use <a href="https://mothur.org/wiki/Silva_reference_files">Silva seed</a> due to its smaller size.</li>
<li>Other classifiers can be found <a href="https://mothur.org/wiki/taxonomy_outline/">here</a>.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="mothur-taxonomic-profiling.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://mothur.s3.us-east-2.amazonaws.com/wiki/silva.seed_v138_1.tgz</span>
<span id="cb17-2"><a href="mothur-taxonomic-profiling.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> xvzf silva.seed_v138_1.tgz</span>
<span id="cb17-3"><a href="mothur-taxonomic-profiling.html#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="pp">*</span>.tgz</span></code></pre></div>
</div>
<div id="mothur-basic-wrapper" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> Mothur basic wrapper<a class="anchor" aria-label="anchor" href="#mothur-basic-wrapper"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Below is a minimal wrapper for sequence processing, classification and taxonomic assignment using <code>mothur</code> pipeline.</li>
<li>For more information refer to <code>mothur</code> <a href="https://mothur.org/wiki/miseq_sop/">MiSeq SOP</a>.</li>
<li>We can save this wrapper as <code>mothur_code.batch</code> and place it in a folder named code.</li>
<li>Then we will run the file on command line as shown below. See the commented lines.</li>
<li>We assume that the input data is gunzipped i.e.Â <strong>.fastq.gz</strong>. If not run <code>gzip *.fastq</code> to compress the files.</li>
</ul>
<div class="tmbalert">
<ol style="list-style-type: decimal">
<li>
Note that the <a href="https://en.wikipedia.org/wiki/Shebang_(Unix)"><code>shebang line</code></a> remind us that we are running the analysis on <code>mothur</code> command line NOT <code>bash</code>. This takes the advantage of using the <strong>current</strong> option to refer to the last saved file.
</li>
<li>
If not generating the mapping file automatically you should make sure that it confer to accepted format as in <code>mothur_mapping_files.tsv</code> described in the previous section. In this demo we use <code>bush.files</code>. This means that all output files will be prefixed with the term <u>bush</u> to reflect the name of the project. Try to avoid loner names.
</li>
</ol>
</div>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="mothur-taxonomic-profiling.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/mothur</span></span>
<span id="cb18-2"><a href="mothur-taxonomic-profiling.html#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="mothur-taxonomic-profiling.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Usage: mothur code/mothur_code.batch # if mothur is in the path</span></span>
<span id="cb18-4"><a href="mothur-taxonomic-profiling.html#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Usage: ./mothur code/mothur_code.batch # on mac/linux if the executable `mothur` is in the working directory.</span></span>
<span id="cb18-5"><a href="mothur-taxonomic-profiling.html#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Usage: ./mothur.exe code/mothur_code.batch # on Windows machins if the executable `mothur.exe` is in the working directory.</span></span>
<span id="cb18-6"><a href="mothur-taxonomic-profiling.html#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="mothur-taxonomic-profiling.html#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="ex">set.current</span><span class="er">(</span><span class="va">processors</span><span class="op">=</span>1<span class="kw">)</span></span>
<span id="cb18-8"><a href="mothur-taxonomic-profiling.html#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co"># set.logfile(name=make_files.logfile)</span></span>
<span id="cb18-9"><a href="mothur-taxonomic-profiling.html#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co"># make.file(inputdir=., type=fastq.gz, prefix=test)</span></span>
<span id="cb18-10"><a href="mothur-taxonomic-profiling.html#cb18-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-11"><a href="mothur-taxonomic-profiling.html#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="ex">set.logfile</span><span class="er">(</span><span class="va">name</span><span class="op">=</span>seq_assembly.logfile<span class="kw">)</span></span>
<span id="cb18-12"><a href="mothur-taxonomic-profiling.html#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="ex">make.contigs</span><span class="er">(</span><span class="va">file</span><span class="op">=</span>bush.files, <span class="va">outputdir</span><span class="op">=</span>./test, <span class="va">maxambig</span><span class="op">=</span>0, <span class="va">maxlength</span><span class="op">=</span>275<span class="kw">);</span></span>
<span id="cb18-13"><a href="mothur-taxonomic-profiling.html#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="ex">unique.seqs</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>current<span class="kw">);</span></span>
<span id="cb18-14"><a href="mothur-taxonomic-profiling.html#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current<span class="kw">)</span></span>
<span id="cb18-15"><a href="mothur-taxonomic-profiling.html#cb18-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-16"><a href="mothur-taxonomic-profiling.html#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="ex">set.logfile</span><span class="er">(</span><span class="va">name</span><span class="op">=</span>seq_align_preclustering.logfile<span class="kw">)</span></span>
<span id="cb18-17"><a href="mothur-taxonomic-profiling.html#cb18-17" aria-hidden="true" tabindex="-1"></a><span class="ex">align.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">reference</span><span class="op">=</span>silva.seed_v138_1.align<span class="kw">);</span></span>
<span id="cb18-18"><a href="mothur-taxonomic-profiling.html#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="ex">screen.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">start</span><span class="op">=</span>13862, <span class="va">end</span><span class="op">=</span>23444, <span class="va">maxhomop</span><span class="op">=</span>8<span class="kw">);</span></span>
<span id="cb18-19"><a href="mothur-taxonomic-profiling.html#cb18-19" aria-hidden="true" tabindex="-1"></a><span class="ex">filter.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">vertical</span><span class="op">=</span>T, <span class="va">trump</span><span class="op">=</span>.<span class="kw">);</span></span>
<span id="cb18-20"><a href="mothur-taxonomic-profiling.html#cb18-20" aria-hidden="true" tabindex="-1"></a><span class="ex">pre.cluster</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">diffs</span><span class="op">=</span>2<span class="kw">);</span></span>
<span id="cb18-21"><a href="mothur-taxonomic-profiling.html#cb18-21" aria-hidden="true" tabindex="-1"></a><span class="ex">unique.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current<span class="kw">);</span></span>
<span id="cb18-22"><a href="mothur-taxonomic-profiling.html#cb18-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-23"><a href="mothur-taxonomic-profiling.html#cb18-23" aria-hidden="true" tabindex="-1"></a><span class="ex">set.logfile</span><span class="er">(</span><span class="va">name</span><span class="op">=</span>chimera_removal.logfile<span class="kw">)</span></span>
<span id="cb18-24"><a href="mothur-taxonomic-profiling.html#cb18-24" aria-hidden="true" tabindex="-1"></a><span class="ex">chimera.vsearch</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">dereplicate</span><span class="op">=</span>t<span class="kw">);</span></span>
<span id="cb18-25"><a href="mothur-taxonomic-profiling.html#cb18-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-26"><a href="mothur-taxonomic-profiling.html#cb18-26" aria-hidden="true" tabindex="-1"></a><span class="ex">set.logfile</span><span class="er">(</span><span class="va">name</span><span class="op">=</span>silva_seed_classification.logfile<span class="kw">)</span></span>
<span id="cb18-27"><a href="mothur-taxonomic-profiling.html#cb18-27" aria-hidden="true" tabindex="-1"></a><span class="ex">classify.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">reference</span><span class="op">=</span>silva.seed_v138_1.align, <span class="va">taxonomy</span><span class="op">=</span>silva.seed_v138_1.tax, <span class="va">cutoff</span><span class="op">=</span>100<span class="kw">);</span></span>
<span id="cb18-28"><a href="mothur-taxonomic-profiling.html#cb18-28" aria-hidden="true" tabindex="-1"></a><span class="ex">remove.lineage</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">taxonomy</span><span class="op">=</span>current, <span class="va">taxon</span><span class="op">=</span>Chloroplast-Mitochondria-unknown-Archaea-Eukaryota<span class="kw">);</span></span>
<span id="cb18-29"><a href="mothur-taxonomic-profiling.html#cb18-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-30"><a href="mothur-taxonomic-profiling.html#cb18-30" aria-hidden="true" tabindex="-1"></a><span class="ex">set.logfile</span><span class="er">(</span><span class="va">name</span><span class="op">=</span>final_files.logfile<span class="kw">)</span></span>
<span id="cb18-31"><a href="mothur-taxonomic-profiling.html#cb18-31" aria-hidden="true" tabindex="-1"></a><span class="ex">rename.file</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">taxonomy</span><span class="op">=</span>current, <span class="va">prefix</span><span class="op">=</span>final<span class="kw">)</span></span>
<span id="cb18-32"><a href="mothur-taxonomic-profiling.html#cb18-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-33"><a href="mothur-taxonomic-profiling.html#cb18-33" aria-hidden="true" tabindex="-1"></a><span class="ex">set.logfile</span><span class="er">(</span><span class="va">name</span><span class="op">=</span>otu_clustering.logfile<span class="kw">)</span></span>
<span id="cb18-34"><a href="mothur-taxonomic-profiling.html#cb18-34" aria-hidden="true" tabindex="-1"></a><span class="ex">dist.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">cutoff</span><span class="op">=</span>0.03<span class="kw">);</span></span>
<span id="cb18-35"><a href="mothur-taxonomic-profiling.html#cb18-35" aria-hidden="true" tabindex="-1"></a><span class="ex">cluster</span><span class="er">(</span><span class="va">column</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">cutoff</span><span class="op">=</span>0.03<span class="kw">);</span></span>
<span id="cb18-36"><a href="mothur-taxonomic-profiling.html#cb18-36" aria-hidden="true" tabindex="-1"></a><span class="ex">make.shared</span><span class="er">(</span><span class="va">list</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">label</span><span class="op">=</span>0.03<span class="kw">);</span></span>
<span id="cb18-37"><a href="mothur-taxonomic-profiling.html#cb18-37" aria-hidden="true" tabindex="-1"></a><span class="ex">classify.otu</span><span class="er">(</span><span class="va">list</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">taxonomy</span><span class="op">=</span>current, <span class="va">label</span><span class="op">=</span>0.03<span class="kw">);</span></span>
<span id="cb18-38"><a href="mothur-taxonomic-profiling.html#cb18-38" aria-hidden="true" tabindex="-1"></a><span class="ex">make.lefse</span><span class="er">(</span><span class="va">shared</span><span class="op">=</span>current, <span class="va">constaxonomy</span><span class="op">=</span>current<span class="kw">);</span></span>
<span id="cb18-39"><a href="mothur-taxonomic-profiling.html#cb18-39" aria-hidden="true" tabindex="-1"></a><span class="ex">make.biom</span><span class="er">(</span><span class="va">shared</span><span class="op">=</span>current, <span class="va">constaxonomy</span><span class="op">=</span>current<span class="kw">);</span></span>
<span id="cb18-40"><a href="mothur-taxonomic-profiling.html#cb18-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-41"><a href="mothur-taxonomic-profiling.html#cb18-41" aria-hidden="true" tabindex="-1"></a><span class="ex">set.logfile</span><span class="er">(</span><span class="va">name</span><span class="op">=</span>phylotype_clustering.logfile<span class="kw">)</span></span>
<span id="cb18-42"><a href="mothur-taxonomic-profiling.html#cb18-42" aria-hidden="true" tabindex="-1"></a><span class="ex">phylotype</span><span class="er">(</span><span class="va">taxonomy</span><span class="op">=</span>current<span class="kw">);</span></span>
<span id="cb18-43"><a href="mothur-taxonomic-profiling.html#cb18-43" aria-hidden="true" tabindex="-1"></a><span class="ex">make.shared</span><span class="er">(</span><span class="va">list</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">label</span><span class="op">=</span>1<span class="kw">);</span></span>
<span id="cb18-44"><a href="mothur-taxonomic-profiling.html#cb18-44" aria-hidden="true" tabindex="-1"></a><span class="ex">classify.otu</span><span class="er">(</span><span class="va">list</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">taxonomy</span><span class="op">=</span>current, <span class="va">label</span><span class="op">=</span>1<span class="kw">);</span></span>
<span id="cb18-45"><a href="mothur-taxonomic-profiling.html#cb18-45" aria-hidden="true" tabindex="-1"></a><span class="ex">make.lefse</span><span class="er">(</span><span class="va">shared</span><span class="op">=</span>current, <span class="va">constaxonomy</span><span class="op">=</span>current<span class="kw">);</span></span>
<span id="cb18-46"><a href="mothur-taxonomic-profiling.html#cb18-46" aria-hidden="true" tabindex="-1"></a><span class="ex">make.biom</span><span class="er">(</span><span class="va">shared</span><span class="op">=</span>current, <span class="va">constaxonomy</span><span class="op">=</span>current<span class="kw">);</span></span>
<span id="cb18-47"><a href="mothur-taxonomic-profiling.html#cb18-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-48"><a href="mothur-taxonomic-profiling.html#cb18-48" aria-hidden="true" tabindex="-1"></a><span class="ex">set.logfile</span><span class="er">(</span><span class="va">name</span><span class="op">=</span>asv_clustering.logfile<span class="kw">)</span></span>
<span id="cb18-49"><a href="mothur-taxonomic-profiling.html#cb18-49" aria-hidden="true" tabindex="-1"></a><span class="ex">make.shared</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>current<span class="kw">)</span></span>
<span id="cb18-50"><a href="mothur-taxonomic-profiling.html#cb18-50" aria-hidden="true" tabindex="-1"></a><span class="ex">classify.otu</span><span class="er">(</span><span class="va">list</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">taxonomy</span><span class="op">=</span>current, <span class="va">label</span><span class="op">=</span>ASV<span class="kw">)</span></span>
<span id="cb18-51"><a href="mothur-taxonomic-profiling.html#cb18-51" aria-hidden="true" tabindex="-1"></a><span class="ex">make.lefse</span><span class="er">(</span><span class="va">shared</span><span class="op">=</span>current, <span class="va">constaxonomy</span><span class="op">=</span>current<span class="kw">)</span></span>
<span id="cb18-52"><a href="mothur-taxonomic-profiling.html#cb18-52" aria-hidden="true" tabindex="-1"></a><span class="ex">make.biom</span><span class="er">(</span><span class="va">shared</span><span class="op">=</span>current, <span class="va">constaxonomy</span><span class="op">=</span>current<span class="kw">)</span></span>
<span id="cb18-53"><a href="mothur-taxonomic-profiling.html#cb18-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-54"><a href="mothur-taxonomic-profiling.html#cb18-54" aria-hidden="true" tabindex="-1"></a><span class="ex">set.logfile</span><span class="er">(</span><span class="va">name</span><span class="op">=</span>phylogenetic_clustering.logfile<span class="kw">)</span></span>
<span id="cb18-55"><a href="mothur-taxonomic-profiling.html#cb18-55" aria-hidden="true" tabindex="-1"></a><span class="ex">dist.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">output</span><span class="op">=</span>lt<span class="kw">)</span></span>
<span id="cb18-56"><a href="mothur-taxonomic-profiling.html#cb18-56" aria-hidden="true" tabindex="-1"></a><span class="ex">clearcut</span><span class="er">(</span><span class="va">phylip</span><span class="op">=</span>current<span class="kw">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="microbial-profiling.html"><span class="header-section-number">3</span> Microbial Profiling</a></div>
<div class="next"><a href="qiime2-taxonomic-profiling.html"><span class="header-section-number">5</span> Mcrobial Profiling Using qiime2 Pipeline</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#mothur-taxonomic-profiling"><span class="header-section-number">4</span> Mcrobial profiling Using mothur Pipeline</a></li>
<li><a class="nav-link" href="#download-a-mothur-trained-classifer"><span class="header-section-number">4.1</span> Download a Mothur trained classifer</a></li>
<li><a class="nav-link" href="#mothur-basic-wrapper"><span class="header-section-number">4.2</span> Mothur basic wrapper</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/CDI2021/smda_part2/blob/main/02b_bioinfomatics_analysis.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/CDI2021/smda_part2/edit/main/02b_bioinfomatics_analysis.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p><strong>Bioinformatics Analysis of Microbiome Data</strong> is PART 2 of the <a class="text-light" href="https://complexdatainsights.com/books/microbiome-analysis-in-R/"><u>End-to-End Practical User Guides</u></a> written by Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA. 
  </p>
</div>

  <div class="col-12 col-md-6 mt-3">
    <p>This guide was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div>
</div>
</footer>
</body>
</html>
