<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 1 Preprocessing 16S rRNA Sequencing Data | Bioinformatics Analysis of Microbiome Data</title>
<meta name="author" content="Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA">
<meta name="description" content="1.1 Initial read quality scores We will use fastqc software to explore the original read qualities before trimming and decontamination. mkdir data/fastqc1 fastqc *.fastq.gz -o data/fastqc1 mkdir...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Chapter 1 Preprocessing 16S rRNA Sequencing Data | Bioinformatics Analysis of Microbiome Data">
<meta property="og:type" content="book">
<meta property="og:description" content="1.1 Initial read quality scores We will use fastqc software to explore the original read qualities before trimming and decontamination. mkdir data/fastqc1 fastqc *.fastq.gz -o data/fastqc1 mkdir...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 1 Preprocessing 16S rRNA Sequencing Data | Bioinformatics Analysis of Microbiome Data">
<meta name="twitter:description" content="1.1 Initial read quality scores We will use fastqc software to explore the original read qualities before trimming and decontamination. mkdir data/fastqc1 fastqc *.fastq.gz -o data/fastqc1 mkdir...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/_Sen-0.4.1/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/_Bitter-0.4.1/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet">
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides">Bioinformatics Analysis of Microbiome Data</a>:
        <small class="text-muted">End-to-End Practical User Guides</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Bioinformatics Analysis of Microbiome Data</a></li>
<li><a class="active" href="preprocessing-16S-reads.html"><span class="header-section-number">1</span> Preprocessing 16S rRNA Sequencing Data</a></li>
<li><a class="" href="preprocess-metag-reads.html"><span class="header-section-number">2</span> Preprocessing Metagenomics Sequencing Data</a></li>
<li><a class="" href="taxonomic-profiling-of-microbiome-data.html"><span class="header-section-number">3</span> Taxonomic Profiling of Microbiome Data</a></li>
<li><a class="" href="mothur-taxonomic-profiling.html"><span class="header-section-number">4</span> Mcrobial profiling Using mothur Pipeline</a></li>
<li><a class="" href="qiime2-taxonomic-profiling.html"><span class="header-section-number">5</span> Mcrobial Profiling Using qiime2 Pipeline</a></li>
<li><a class="" href="metaphlan-taxonomic-profiling.html"><span class="header-section-number">6</span> Taxonomic Profiling of Metagenomics Sequencing Data</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></li>
<li><a class="" href="software-and-packages-used.html"><span class="header-section-number">B</span> Software and Packages Used</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/CDI2021/smda_part2">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="preprocessing-16S-reads" class="section level1" number="1">
<h1>
<span class="header-section-number">1</span> Preprocessing 16S rRNA Sequencing Data<a class="anchor" aria-label="anchor" href="#preprocessing-16S-reads"><i class="fas fa-link"></i></a>
</h1>
<div id="initial-read-quality-scores" class="section level2" number="1.1">
<h2>
<span class="header-section-number">1.1</span> Initial read quality scores<a class="anchor" aria-label="anchor" href="#initial-read-quality-scores"><i class="fas fa-link"></i></a>
</h2>
<p>We will use fastqc software to explore the original read qualities before trimming and decontamination.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="preprocessing-16S-reads.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> data/fastqc1</span>
<span id="cb1-2"><a href="preprocessing-16S-reads.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> <span class="pp">*</span>.fastq.gz <span class="at">-o</span> data/fastqc1</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="preprocessing-16S-reads.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> data/multiqc1</span>
<span id="cb2-2"><a href="preprocessing-16S-reads.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">-f</span> <span class="at">--data-dir</span> data/fastqc1 <span class="at">-o</span> data/multiqc1 <span class="at">--export</span></span></code></pre></div>
<blockquote>
<p>Plots from <code>multiqc</code> are exported to <code>multiqc_plots</code> folder.</p>
</blockquote>
<div class="figure">
<img src="/Users/tmbuza/Dropbox/CDILLC/GIT_REPOS/smda_end2end/data/multiqc1/multiqc_plots/png/mqc_fastqc_per_base_sequence_quality_plot_1.png" style="width:100.0%" alt=""><p class="caption">Original: Mean quality scores</p>
</div>
<p><br></p>
<div class="figure">
<img src="/Users/tmbuza/Dropbox/CDILLC/GIT_REPOS/smda_end2end/data/multiqc1/multiqc_plots/png/mqc_fastqc_per_sequence_quality_scores_plot_1.png" style="width:100.0%" alt=""><p class="caption">Original: Per sequence quality scores</p>
</div>
<p><br></p>
</div>
<div id="read-trimming-using-bbduk.sh-from-bbmap-platform" class="section level2" number="1.2">
<h2>
<span class="header-section-number">1.2</span> Read trimming using <code>bbduk.sh</code> from bbmap platform<a class="anchor" aria-label="anchor" href="#read-trimming-using-bbduk.sh-from-bbmap-platform"><i class="fas fa-link"></i></a>
</h2>
<blockquote>
<p>Hint! Make sure that the pattern in the script is in the file name. Some files may contain <code>R1_001.fastq.gz</code>. Here we use files downloaded from NCBI-SRA, they do not use R1_001 in the sample name.</p>
</blockquote>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="preprocessing-16S-reads.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="kw">`</span><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>_1.fastq.gz <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/_1.fastq.gz//'</span><span class="kw">`</span></span>
<span id="cb3-2"><a href="preprocessing-16S-reads.html#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span></span>
<span id="cb3-3"><a href="preprocessing-16S-reads.html#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">bbduk.sh</span> <span class="at">-Xmx4g</span> in1=<span class="va">$i</span><span class="dt">\_</span>1.fastq.gz in2=<span class="va">$i</span><span class="dt">\_</span>2.fastq.gz out1=data/trimmed/<span class="va">$i</span><span class="dt">\_</span>1.fastq.gz out2=data/trimmed/<span class="va">$i</span><span class="dt">\_</span>2.fastq.gz qtrim=r trimq=25 overwrite=True</span>
<span id="cb3-4"><a href="preprocessing-16S-reads.html#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="preprocessing-16S-reads.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> data/stats2  </span>
<span id="cb4-2"><a href="preprocessing-16S-reads.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> stat data/trimmed/<span class="pp">*</span>.fastq.gz <span class="op">&gt;</span>data/stats2/seqkit_stats.txt</span>
<span id="cb4-3"><a href="preprocessing-16S-reads.html#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="preprocessing-16S-reads.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> data/fastqc2</span>
<span id="cb4-5"><a href="preprocessing-16S-reads.html#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> data/trimmed/<span class="pp">*</span>.fastq.gz <span class="at">-o</span> data/fastqc2</span>
<span id="cb4-6"><a href="preprocessing-16S-reads.html#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="preprocessing-16S-reads.html#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> data/multiqc2</span>
<span id="cb4-8"><a href="preprocessing-16S-reads.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">-f</span> <span class="at">--data-dir</span> data/fastqc2 <span class="at">-o</span> data/multiqc2 <span class="at">--export</span></span></code></pre></div>
<div class="figure">
<img src="/Users/tmbuza/Dropbox/CDILLC/GIT_REPOS/smda_end2end/data/multiqc2/multiqc_plots/png/mqc_fastqc_per_base_sequence_quality_plot_1.png" style="width:100.0%" alt=""><p class="caption">Trimmed: Mean quality scores</p>
</div>
<p><br></p>
<div class="figure">
<img src="/Users/tmbuza/Dropbox/CDILLC/GIT_REPOS/smda_end2end/data/multiqc2/multiqc_plots/png/mqc_fastqc_per_sequence_quality_scores_plot_1.png" style="width:100.0%" alt=""><p class="caption">Trimmed: Per sequence quality scores</p>
</div>
</div>
<div id="read-decontamination" class="section level2" number="1.3">
<h2>
<span class="header-section-number">1.3</span> Read decontamination<a class="anchor" aria-label="anchor" href="#read-decontamination"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Using <code>bbduk.sh</code> from bbmap platform</li>
<li>This remove some contamination, e.g. PhiX Control reads.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="preprocessing-16S-reads.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="kw">`</span><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>_1.fastq.gz <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/_1.fastq.gz//'</span><span class="kw">`</span></span>
<span id="cb5-2"><a href="preprocessing-16S-reads.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb5-3"><a href="preprocessing-16S-reads.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="ex">bbduk.sh</span> <span class="at">-Xmx4g</span> in1=data/trimmed/<span class="va">$i</span><span class="dt">\_</span>1.fastq.gz in2=data/trimmed/<span class="va">$i</span><span class="dt">\_</span>2.fastq.gz out1=data/decontam/<span class="va">$i</span><span class="dt">\_</span>1.fastq.gz out2=data/decontam/<span class="va">$i</span><span class="dt">\_</span>2.fastq.gz outm1=data/decontam/matchedphix/<span class="va">$i</span><span class="dt">\_</span>1.fastq.gz outm2=data/decontam/matchedphix/<span class="va">$i</span><span class="dt">\_</span>2.fastq.gz ref=~/bbmap/resources/phix174_ill.ref.fa.gz k=31 hdist=1 overwrite=True</span>
<span id="cb5-4"><a href="preprocessing-16S-reads.html#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="preprocessing-16S-reads.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> data/stats3  </span>
<span id="cb6-2"><a href="preprocessing-16S-reads.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> stat data/decontam/<span class="pp">*</span>.fastq.gz <span class="op">&gt;</span>data/stats3/seqkit_stats.txt</span>
<span id="cb6-3"><a href="preprocessing-16S-reads.html#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="preprocessing-16S-reads.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> data/fastqc3</span>
<span id="cb6-5"><a href="preprocessing-16S-reads.html#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> data/decontam/<span class="pp">*</span>.fastq.gz <span class="at">-o</span> data/fastqc3</span>
<span id="cb6-6"><a href="preprocessing-16S-reads.html#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="preprocessing-16S-reads.html#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> data/multiqc3</span>
<span id="cb6-8"><a href="preprocessing-16S-reads.html#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">-f</span> <span class="at">--data-dir</span> data/fastqc3 <span class="at">-o</span> data/multiqc3 <span class="at">--export</span></span></code></pre></div>
<p><br></p>
<div class="figure">
<img src="/Users/tmbuza/Dropbox/CDILLC/GIT_REPOS/smda_end2end/data/multiqc3/multiqc_plots/png/mqc_fastqc_per_base_sequence_quality_plot_1.png" style="width:100.0%" alt=""><p class="caption">Decontaminated: Mean quality scores</p>
</div>
<p><br></p>
<div class="figure">
<img src="/Users/tmbuza/Dropbox/CDILLC/GIT_REPOS/smda_end2end/data/multiqc3/multiqc_plots/png/mqc_fastqc_per_sequence_quality_scores_plot_1.png" style="width:100.0%" alt=""><p class="caption">Decontaminated: Per sequence quality scores</p>
</div>
<p><br></p>
</div>
<div id="compare-preprocessed-reads" class="section level2" number="1.4">
<h2>
<span class="header-section-number">1.4</span> Compare preprocessed reads<a class="anchor" aria-label="anchor" href="#compare-preprocessed-reads"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span>, <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/ggtext/">ggtext</a></span><span class="op">)</span>

<span class="va">stats1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_table.html">read_table</a></span><span class="op">(</span><span class="st">"data/stats1/seqkit_stats.txt"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">file</span>, <span class="st">".*/"</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">file</span>, original <span class="op">=</span> <span class="va">num_seqs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">stats1</span>, <span class="st">"RDataRDS/seqkit_stats.rds"</span><span class="op">)</span>

<span class="va">stats2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_table.html">read_table</a></span><span class="op">(</span><span class="st">"data/stats2/seqkit_stats.txt"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">file</span>, <span class="st">".*/"</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">file</span>, trimmed <span class="op">=</span> <span class="va">num_seqs</span><span class="op">)</span>

<span class="va">stats3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_table.html">read_table</a></span><span class="op">(</span><span class="st">"data/stats3/seqkit_stats.txt"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">file</span>, <span class="st">".*/"</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">file</span>, decontaminated <span class="op">=</span> <span class="va">num_seqs</span><span class="op">)</span>

<span class="va">read_status</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">stats1</span>, <span class="va">stats2</span>, by <span class="op">=</span> <span class="st">"file"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">stats3</span>, by <span class="op">=</span> <span class="st">"file"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html">str_detect</a></span><span class="op">(</span><span class="va">file</span>, <span class="st">"_1"</span><span class="op">)</span>, <span class="st">"foward"</span>, <span class="st">"reverse"</span><span class="op">)</span>, .before<span class="op">=</span><span class="va">original</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">file</span>, <span class="va">strand</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"variable"</span>, values_to <span class="op">=</span> <span class="st">"num_seqs"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">read_status</span><span class="op">)</span></code></pre></div>
<pre><code># A tibble: 6 × 4
  file                  strand  variable       num_seqs
  &lt;chr&gt;                 &lt;chr&gt;   &lt;chr&gt;             &lt;dbl&gt;
1 SRR7450706_1.fastq.gz foward  original         383666
2 SRR7450706_1.fastq.gz foward  trimmed          383062
3 SRR7450706_1.fastq.gz foward  decontaminated   358493
4 SRR7450706_2.fastq.gz reverse original         383666
5 SRR7450706_2.fastq.gz reverse trimmed          383062
6 SRR7450706_2.fastq.gz reverse decontaminated   358493</code></pre>
</div>
<div id="what-is-the-distribution-of-the-processed-reads" class="section level2" number="1.5">
<h2>
<span class="header-section-number">1.5</span> What is the distribution of the processed reads<a class="anchor" aria-label="anchor" href="#what-is-the-distribution-of-the-processed-reads"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">read_status</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>variable <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">variable</span><span class="op">)</span>,
         variable <span class="op">=</span> <span class="fu"><a href="https://forcats.tidyverse.org/reference/fct_reorder.html">fct_reorder</a></span><span class="op">(</span><span class="va">variable</span>, <span class="va">num_seqs</span>, .desc<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">strand</span>, y <span class="op">=</span> <span class="va">num_seqs</span><span class="op">/</span><span class="fl">1000</span>, fill <span class="op">=</span> <span class="va">variable</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="st">"Number of Reads (1000)"</span>, fill <span class="op">=</span> <span class="st">"Preprocess"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://wilkelab.org/ggtext/reference/element_markdown.html">element_markdown</a></span><span class="op">(</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>,
        legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">12</span>, <span class="st">"pt"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="va">nowhitespace</span></code></pre></div>
<div class="inline-figure"><img src="figures/unnamed-chunk-2-1.png" width="80%" height="40%" style="display: block; margin: auto;"></div>
<p><br></p>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html">Bioinformatics Analysis of Microbiome Data</a></div>
<div class="next"><a href="preprocess-metag-reads.html"><span class="header-section-number">2</span> Preprocessing Metagenomics Sequencing Data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#preprocessing-16S-reads"><span class="header-section-number">1</span> Preprocessing 16S rRNA Sequencing Data</a></li>
<li><a class="nav-link" href="#initial-read-quality-scores"><span class="header-section-number">1.1</span> Initial read quality scores</a></li>
<li><a class="nav-link" href="#read-trimming-using-bbduk.sh-from-bbmap-platform"><span class="header-section-number">1.2</span> Read trimming using bbduk.sh from bbmap platform</a></li>
<li><a class="nav-link" href="#read-decontamination"><span class="header-section-number">1.3</span> Read decontamination</a></li>
<li><a class="nav-link" href="#compare-preprocessed-reads"><span class="header-section-number">1.4</span> Compare preprocessed reads</a></li>
<li><a class="nav-link" href="#what-is-the-distribution-of-the-processed-reads"><span class="header-section-number">1.5</span> What is the distribution of the processed reads</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/CDI2021/smda_part2/blob/main/02a_read_preprocessing.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/CDI2021/smda_part2/edit/main/02a_read_preprocessing.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p><strong>Bioinformatics Analysis of Microbiome Data</strong> is PART 2 of the <a class="text-light" href="https://complexdatainsights.com/books/microbiome-analysis-in-R/"><u>End-to-End Practical User Guides</u></a> written by Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA. 
  </p>
</div>

  <div class="col-12 col-md-6 mt-3">
    <p>This guide was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div>
</div>
</footer>
</body>
</html>
