<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 2 Preprocessing Metagenomics Sequencing Data | Bioinformatics Analysis of Microbiome Data</title>
<meta name="author" content="Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA">
<meta name="description" content="The kneaddata tool from biobakery biobakery is famous for preprocessing metagenomics data. It integrates several other quality control tools such as FastQC and Trimmomatic. Here we demonstrate how...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Chapter 2 Preprocessing Metagenomics Sequencing Data | Bioinformatics Analysis of Microbiome Data">
<meta property="og:type" content="book">
<meta property="og:description" content="The kneaddata tool from biobakery biobakery is famous for preprocessing metagenomics data. It integrates several other quality control tools such as FastQC and Trimmomatic. Here we demonstrate how...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 2 Preprocessing Metagenomics Sequencing Data | Bioinformatics Analysis of Microbiome Data">
<meta name="twitter:description" content="The kneaddata tool from biobakery biobakery is famous for preprocessing metagenomics data. It integrates several other quality control tools such as FastQC and Trimmomatic. Here we demonstrate how...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/_Sen-0.4.1/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/_Bitter-0.4.1/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides">Bioinformatics Analysis of Microbiome Data</a>:
        <small class="text-muted">End-to-End Practical User Guides</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Bioinformatics Analysis of Microbiome Data</a></li>
<li class="book-part">BIOINFORMATICS ANALYSIS</li>
<li><a class="" href="preprocess-16S-reads.html"><span class="header-section-number">1</span> Sequence Preprocessing</a></li>
<li><a class="active" href="preprocess-metag-reads.html"><span class="header-section-number">2</span> Preprocessing Metagenomics Sequencing Data</a></li>
<li><a class="" href="microbial-profiling.html"><span class="header-section-number">3</span> Microbial Profiling</a></li>
<li><a class="" href="mothur-taxonomic-profiling.html"><span class="header-section-number">4</span> Mcrobial profiling Using mothur Pipeline</a></li>
<li><a class="" href="qiime2-taxonomic-profiling.html"><span class="header-section-number">5</span> Mcrobial Profiling Using qiime2 Pipeline</a></li>
<li><a class="" href="metaphlan-taxonomic-profiling.html"><span class="header-section-number">6</span> Taxonomic Profiling of Metagenomics Sequencing Data</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></li>
<li><a class="" href="software-and-packages.html"><span class="header-section-number">B</span> Software and Packages</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/CDI2021/smda_part2">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="preprocess-metag-reads" class="section level1" number="2">
<h1>
<span class="header-section-number">2</span> Preprocessing Metagenomics Sequencing Data<a class="anchor" aria-label="anchor" href="#preprocess-metag-reads"><i class="fas fa-link"></i></a>
</h1>
<p>The <code>kneaddata</code> tool from biobakery <a href="https://github.com/biobakery/kneaddata">biobakery</a> is famous for preprocessing metagenomics data. It integrates several other quality control tools such as FastQC and Trimmomatic. Here we demonstrate how to use SeqKit, kneaddata and MultiQC t preprocess metagenomics sequencing data.</p>
<div id="what-are-the-read-statistics" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> What are the read statistics<a class="anchor" aria-label="anchor" href="#what-are-the-read-statistics"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Using <code>seqkit stat</code> function.</li>
<li>Run the function on command line.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="preprocess-metag-reads.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> kneaddata</span>
<span id="cb9-2"><a href="preprocess-metag-reads.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> kneaddata/stats1  </span>
<span id="cb9-3"><a href="preprocess-metag-reads.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> stat <span class="pp">*</span>.fastq.gz <span class="op">&gt;</span>kneaddata/stats1/seqkit_stats.txt</span></code></pre></div>
</div>
<div id="initial-read-quality-scores-1" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> Initial read quality scores<a class="anchor" aria-label="anchor" href="#initial-read-quality-scores-1"><i class="fas fa-link"></i></a>
</h2>
<p>The <code>kneaddata</code> from biobakery platform is exclusively used to preprocess the metagenomics data.
- Here we set the <code>kneaddata</code> to run <code>fastqc</code> as the first step using <code>run-fastqc-start</code> function.
- Then <code>fastqc</code> results are then summarized outside <code>kneaddata</code> pipeline using <code>multiqc</code> software.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="preprocess-metag-reads.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> data/<span class="pp">*</span>.fastq.gz</span>
<span id="cb10-2"><a href="preprocess-metag-reads.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb10-3"><a href="preprocess-metag-reads.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> kneaddata <span class="at">--i</span> <span class="va">$i</span> <span class="dt">\</span></span>
<span id="cb10-4"><a href="preprocess-metag-reads.html#cb10-4" aria-hidden="true" tabindex="-1"></a>-o kneaddata/fastqc1 <span class="dt">\</span></span>
<span id="cb10-5"><a href="preprocess-metag-reads.html#cb10-5" aria-hidden="true" tabindex="-1"></a>--run-fastqc-start <span class="dt">\</span></span>
<span id="cb10-6"><a href="preprocess-metag-reads.html#cb10-6" aria-hidden="true" tabindex="-1"></a>--bypass-trim <span class="dt">\</span></span>
<span id="cb10-7"><a href="preprocess-metag-reads.html#cb10-7" aria-hidden="true" tabindex="-1"></a>--bypass-trf <span class="dt">\</span></span>
<span id="cb10-8"><a href="preprocess-metag-reads.html#cb10-8" aria-hidden="true" tabindex="-1"></a>--sequencer-source <span class="st">"none"</span> <span class="dt">\</span></span>
<span id="cb10-9"><a href="preprocess-metag-reads.html#cb10-9" aria-hidden="true" tabindex="-1"></a>-t 4 </span>
<span id="cb10-10"><a href="preprocess-metag-reads.html#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb10-11"><a href="preprocess-metag-reads.html#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="preprocess-metag-reads.html#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="preprocess-metag-reads.html#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> kneaddata/multiqc1</span>
<span id="cb10-14"><a href="preprocess-metag-reads.html#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">-f</span> <span class="at">--data-dir</span> kneaddata/fastqc1 <span class="at">-o</span> kneaddata/multiqc1 <span class="at">--export</span></span></code></pre></div>
</div>
<div id="read-trimming" class="section level2" number="2.3">
<h2>
<span class="header-section-number">2.3</span> Read Trimming<a class="anchor" aria-label="anchor" href="#read-trimming"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Using trimmomatic tool to trim the reads using the specified options.</li>
<li>The fastqc function is run after trimming</li>
<li>Then fastqc results are summarized using multiqc software.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="preprocess-metag-reads.html#cb11-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-2"><a href="preprocess-metag-reads.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> data/<span class="pp">*</span>.fastq.gz</span>
<span id="cb11-3"><a href="preprocess-metag-reads.html#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb11-4"><a href="preprocess-metag-reads.html#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="bu">time</span> kneaddata <span class="at">--i</span> <span class="va">$i</span> <span class="dt">\</span></span>
<span id="cb11-5"><a href="preprocess-metag-reads.html#cb11-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">-o</span> kneaddata/fastqc2 <span class="dt">\</span></span>
<span id="cb11-6"><a href="preprocess-metag-reads.html#cb11-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">--trimmomatic</span> /Users/tmbuza/opt/anaconda3/envs/biobakery3/bin/ <span class="dt">\</span></span>
<span id="cb11-7"><a href="preprocess-metag-reads.html#cb11-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">--trimmomatic-options</span> <span class="dt">\</span></span>
<span id="cb11-8"><a href="preprocess-metag-reads.html#cb11-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"ILLUMINACLIP:/Volumes/SeagateTMB/trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 </span><span class="dt">\</span></span>
<span id="cb11-9"><a href="preprocess-metag-reads.html#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="st">        LEADING:3 </span><span class="dt">\</span></span>
<span id="cb11-10"><a href="preprocess-metag-reads.html#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="st">        TRAILING:3 </span><span class="dt">\</span></span>
<span id="cb11-11"><a href="preprocess-metag-reads.html#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="st">        SLIDINGWINDOW:4:20 </span><span class="dt">\</span></span>
<span id="cb11-12"><a href="preprocess-metag-reads.html#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="st">        MINLEN:60"</span> <span class="dt">\</span></span>
<span id="cb11-13"><a href="preprocess-metag-reads.html#cb11-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">--sequencer-source</span> <span class="st">"NexteraPE"</span> <span class="dt">\</span></span>
<span id="cb11-14"><a href="preprocess-metag-reads.html#cb11-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">--run-fastqc-end</span> <span class="dt">\</span></span>
<span id="cb11-15"><a href="preprocess-metag-reads.html#cb11-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">--bypass-trf</span> <span class="dt">\</span></span>
<span id="cb11-16"><a href="preprocess-metag-reads.html#cb11-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">-t</span> 4 </span>
<span id="cb11-17"><a href="preprocess-metag-reads.html#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="preprocess-metag-reads.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> kneaddata/stats2</span>
<span id="cb12-2"><a href="preprocess-metag-reads.html#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> stat kneaddata/fastqc2/<span class="pp">*</span>trimmed.fastq <span class="op">&gt;</span>kneaddata/stats2/seqkit_stats.txt</span>
<span id="cb12-3"><a href="preprocess-metag-reads.html#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="preprocess-metag-reads.html#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> kneaddata/fastqc2/multiqc2</span>
<span id="cb12-5"><a href="preprocess-metag-reads.html#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">-f</span> <span class="at">--data-dir</span> kneaddata/fastqc2 <span class="at">-o</span> kneaddata_fastqc1/multiqc1 <span class="at">--export</span></span></code></pre></div>
</div>
<div id="read-decontamination-1" class="section level2" number="2.4">
<h2>
<span class="header-section-number">2.4</span> Read Decontamination<a class="anchor" aria-label="anchor" href="#read-decontamination-1"><i class="fas fa-link"></i></a>
</h2>
<div id="option-1-using-bowtie2-database" class="section level3" number="2.4.1">
<h3>
<span class="header-section-number">2.4.1</span> Option 1: Using Bowtie2 database<a class="anchor" aria-label="anchor" href="#option-1-using-bowtie2-database"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>Trimming is done using trimmomatic</li>
<li>Then decontamination is done using the bowtie2 database.</li>
<li>Then after decontamination fastqc is implemented to assess the read quality scores of the remaining reads.</li>
<li>Then fastqc results are summarized using multiqc software.</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="preprocess-metag-reads.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> data/<span class="pp">*</span>.fastq.gz</span>
<span id="cb13-2"><a href="preprocess-metag-reads.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb13-3"><a href="preprocess-metag-reads.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> kneaddata <span class="at">--i</span> <span class="va">$i</span> <span class="dt">\</span></span>
<span id="cb13-4"><a href="preprocess-metag-reads.html#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">-o</span> kneaddata_fastqc3 <span class="dt">\</span></span>
<span id="cb13-5"><a href="preprocess-metag-reads.html#cb13-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">--reference-db</span> /Volumes/SeagateTMB/kneaddata_database/ <span class="dt">\</span></span>
<span id="cb13-6"><a href="preprocess-metag-reads.html#cb13-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">--trimmomatic</span> /Users/tmbuza/opt/anaconda3/envs/biobakery3/bin/ <span class="dt">\</span></span>
<span id="cb13-7"><a href="preprocess-metag-reads.html#cb13-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">--trimmomatic-options</span> <span class="dt">\</span></span>
<span id="cb13-8"><a href="preprocess-metag-reads.html#cb13-8" aria-hidden="true" tabindex="-1"></a>          <span class="st">"ILLUMINACLIP:/Volumes/SeagateTMB/trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 </span><span class="dt">\</span></span>
<span id="cb13-9"><a href="preprocess-metag-reads.html#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="st">          LEADING:3 </span><span class="dt">\</span></span>
<span id="cb13-10"><a href="preprocess-metag-reads.html#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="st">          TRAILING:3 </span><span class="dt">\</span></span>
<span id="cb13-11"><a href="preprocess-metag-reads.html#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="st">          SLIDINGWINDOW:4:20 </span><span class="dt">\</span></span>
<span id="cb13-12"><a href="preprocess-metag-reads.html#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="st">          MINLEN:60"</span> <span class="dt">\</span></span>
<span id="cb13-13"><a href="preprocess-metag-reads.html#cb13-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">--sequencer-source</span> <span class="st">"NexteraPE"</span> <span class="dt">\</span></span>
<span id="cb13-14"><a href="preprocess-metag-reads.html#cb13-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">--run-trf</span> <span class="dt">\</span></span>
<span id="cb13-15"><a href="preprocess-metag-reads.html#cb13-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">--run-fastqc-end</span> <span class="dt">\</span></span>
<span id="cb13-16"><a href="preprocess-metag-reads.html#cb13-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">-t</span> 4 </span>
<span id="cb13-17"><a href="preprocess-metag-reads.html#cb13-17" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="preprocess-metag-reads.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> kneaddata/stats3</span>
<span id="cb14-2"><a href="preprocess-metag-reads.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> stat kneaddata/fastqc3/<span class="pp">*</span>trimmed.fastq <span class="op">&gt;</span>kneaddata/stats3/seqkit_stats.txt</span>
<span id="cb14-3"><a href="preprocess-metag-reads.html#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="preprocess-metag-reads.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> kneaddata/fastqc3/multiqc3</span>
<span id="cb14-5"><a href="preprocess-metag-reads.html#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">-f</span> <span class="at">--data-dir</span> kneaddata/fastqc3 <span class="at">-o</span> kneaddata/fastqc3/multiqc3 <span class="at">--export</span></span></code></pre></div>
</div>
<div id="option-2-using-bmtagger-database" class="section level3" number="2.4.2">
<h3>
<span class="header-section-number">2.4.2</span> Option 2: Using BMTagger database<a class="anchor" aria-label="anchor" href="#option-2-using-bmtagger-database"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>Trimming is done using trimmomatic</li>
<li>Then decontamination is done using the BMTagger database.</li>
<li>Then after decontamination fastqc is implemented to assess the read quality scores of the remaining reads.</li>
<li>Then fastqc results are summarized using multiqc software.</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="preprocess-metag-reads.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> data/<span class="pp">*</span>.fastq.gz</span>
<span id="cb15-2"><a href="preprocess-metag-reads.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb15-3"><a href="preprocess-metag-reads.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> kneaddata <span class="at">--i</span> <span class="va">$i</span> <span class="dt">\</span></span>
<span id="cb15-4"><a href="preprocess-metag-reads.html#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">-o</span> kneaddata_fastqc4 <span class="dt">\</span></span>
<span id="cb15-5"><a href="preprocess-metag-reads.html#cb15-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">--reference-db</span> /Volumes/SeagateTMB/Human_Assembly19_BMTagger_DB/ <span class="dt">\</span></span>
<span id="cb15-6"><a href="preprocess-metag-reads.html#cb15-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">--trimmomatic</span> /Users/tmbuza/opt/anaconda3/envs/biobakery3/bin/ <span class="dt">\</span></span>
<span id="cb15-7"><a href="preprocess-metag-reads.html#cb15-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">--trimmomatic-options</span> <span class="dt">\</span></span>
<span id="cb15-8"><a href="preprocess-metag-reads.html#cb15-8" aria-hidden="true" tabindex="-1"></a>          <span class="st">"ILLUMINACLIP:/Volumes/SeagateTMB/trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 </span><span class="dt">\</span></span>
<span id="cb15-9"><a href="preprocess-metag-reads.html#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="st">          LEADING:3 </span><span class="dt">\</span></span>
<span id="cb15-10"><a href="preprocess-metag-reads.html#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="st">          TRAILING:3 </span><span class="dt">\</span></span>
<span id="cb15-11"><a href="preprocess-metag-reads.html#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="st">          SLIDINGWINDOW:4:20 </span><span class="dt">\</span></span>
<span id="cb15-12"><a href="preprocess-metag-reads.html#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="st">          MINLEN:60"</span> <span class="dt">\</span></span>
<span id="cb15-13"><a href="preprocess-metag-reads.html#cb15-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">--sequencer-source</span> <span class="st">"NexteraPE"</span> <span class="dt">\</span></span>
<span id="cb15-14"><a href="preprocess-metag-reads.html#cb15-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">--run-trf</span> <span class="dt">\</span></span>
<span id="cb15-15"><a href="preprocess-metag-reads.html#cb15-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">--run-bmtagger</span> <span class="dt">\</span></span>
<span id="cb15-16"><a href="preprocess-metag-reads.html#cb15-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">--run-fastqc-end</span> <span class="dt">\</span></span>
<span id="cb15-17"><a href="preprocess-metag-reads.html#cb15-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">-t</span> 4 </span>
<span id="cb15-18"><a href="preprocess-metag-reads.html#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="preprocess-metag-reads.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> kneaddata/stats4</span>
<span id="cb16-2"><a href="preprocess-metag-reads.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> stat kneaddata/fastqc4/<span class="pp">*</span>trimmed.fastq <span class="op">&gt;</span>kneaddata/stats4/seqkit_stats.txt</span>
<span id="cb16-3"><a href="preprocess-metag-reads.html#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="preprocess-metag-reads.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> kneaddata/fastqc4/multiqc4</span>
<span id="cb16-5"><a href="preprocess-metag-reads.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">-f</span> <span class="at">--data-dir</span> kneaddata/fastqc4 <span class="at">-o</span> kneaddata/fastqc4/multiqc4 <span class="at">--export</span></span></code></pre></div>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="preprocess-16S-reads.html"><span class="header-section-number">1</span> Sequence Preprocessing</a></div>
<div class="next"><a href="microbial-profiling.html"><span class="header-section-number">3</span> Microbial Profiling</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#preprocess-metag-reads"><span class="header-section-number">2</span> Preprocessing Metagenomics Sequencing Data</a></li>
<li><a class="nav-link" href="#what-are-the-read-statistics"><span class="header-section-number">2.1</span> What are the read statistics</a></li>
<li><a class="nav-link" href="#initial-read-quality-scores-1"><span class="header-section-number">2.2</span> Initial read quality scores</a></li>
<li><a class="nav-link" href="#read-trimming"><span class="header-section-number">2.3</span> Read Trimming</a></li>
<li>
<a class="nav-link" href="#read-decontamination-1"><span class="header-section-number">2.4</span> Read Decontamination</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#option-1-using-bowtie2-database"><span class="header-section-number">2.4.1</span> Option 1: Using Bowtie2 database</a></li>
<li><a class="nav-link" href="#option-2-using-bmtagger-database"><span class="header-section-number">2.4.2</span> Option 2: Using BMTagger database</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/CDI2021/smda_part2/blob/main/02a_read_preprocessing.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/CDI2021/smda_part2/edit/main/02a_read_preprocessing.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p><strong>Bioinformatics Analysis of Microbiome Data</strong> is PART 2 of the <a class="text-light" href="https://complexdatainsights.com/books/microbiome-analysis-in-R/"><u>End-to-End Practical User Guides</u></a> written by Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA. 
  </p>
</div>

  <div class="col-12 col-md-6 mt-3">
    <p>This guide was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div>
</div>
</footer>
</body>
</html>
