<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 5 Mcrobial Profiling Using qiime2 Pipeline | Bioinformatics Analysis of Microbiome Data</title>
<meta name="author" content="Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA">
<meta name="description" content="5.0.1 Download a QIIME2 trained classifer You can use Naive Bayes (nb) classifiers trained on GreenGenes or SILVA database with 99% OTUs. You can train your own classifier using the...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Chapter 5 Mcrobial Profiling Using qiime2 Pipeline | Bioinformatics Analysis of Microbiome Data">
<meta property="og:type" content="book">
<meta property="og:description" content="5.0.1 Download a QIIME2 trained classifer You can use Naive Bayes (nb) classifiers trained on GreenGenes or SILVA database with 99% OTUs. You can train your own classifier using the...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 5 Mcrobial Profiling Using qiime2 Pipeline | Bioinformatics Analysis of Microbiome Data">
<meta name="twitter:description" content="5.0.1 Download a QIIME2 trained classifer You can use Naive Bayes (nb) classifiers trained on GreenGenes or SILVA database with 99% OTUs. You can train your own classifier using the...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/_Sen-0.4.1/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/_Bitter-0.4.1/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
<link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides">Bioinformatics Analysis of Microbiome Data</a>:
        <small class="text-muted">End-to-End Practical User Guides</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Bioinformatics Analysis of Microbiome Data</a></li>
<li class="book-part">BIOINFORMATICS ANALYSIS</li>
<li><a class="" href="preprocess-16S-reads.html"><span class="header-section-number">1</span> Standard Sequence Preprocessing</a></li>
<li><a class="" href="preprocess-kneaddata.html"><span class="header-section-number">2</span> Preprocessing Metagenomics Data Using Kneaddata</a></li>
<li><a class="" href="microbial-profiling.html"><span class="header-section-number">3</span> Microbial Profiling</a></li>
<li><a class="" href="mothur-taxonomic-profiling.html"><span class="header-section-number">4</span> Mcrobial profiling Using mothur Pipeline</a></li>
<li><a class="active" href="qiime2-taxonomic-profiling.html"><span class="header-section-number">5</span> Mcrobial Profiling Using qiime2 Pipeline</a></li>
<li><a class="" href="metaphlan-taxonomic-profiling.html"><span class="header-section-number">6</span> Taxonomic Profiling of Metagenomics Sequencing Data</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></li>
<li><a class="" href="software-and-packages.html"><span class="header-section-number">B</span> Software and Packages</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/CDI2021/smda_part2">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="qiime2-taxonomic-profiling" class="section level1" number="5">
<h1>
<span class="header-section-number">5</span> Mcrobial Profiling Using <code>qiime2</code> Pipeline<a class="anchor" aria-label="anchor" href="#qiime2-taxonomic-profiling"><i class="fas fa-link"></i></a>
</h1>
<div id="download-a-qiime2-trained-classifer" class="section level3" number="5.0.1">
<h3>
<span class="header-section-number">5.0.1</span> Download a QIIME2 trained classifer<a class="anchor" aria-label="anchor" href="#download-a-qiime2-trained-classifer"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>You can use Naive Bayes (nb) classifiers trained on GreenGenes or SILVA database with 99% OTUs. You can train your own classifier using the <a href="https://github.com/qiime2/q2-feature-classifier">q2-feature-classifier</a>.</li>
</ul>
<p>Here we will download the smallest classifier, the naive classifier trained on <a href="https://docs.qiime2.org/2022.2/data-resources/">Greengenes 13_8 99% OTUs from 515F/806R region of sequences</a></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="qiime2-taxonomic-profiling.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="dt">\</span></span>
<span id="cb19-2"><a href="qiime2-taxonomic-profiling.html#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">-O</span> <span class="st">"gg-13-8-99-515-806-nb-classifier.qza"</span> <span class="dt">\</span></span>
<span id="cb19-3"><a href="qiime2-taxonomic-profiling.html#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"https://data.qiime2.org/2022.2/common/gg-13-8-99-515-806-nb-classifier.qza"</span></span></code></pre></div>
</div>
<div id="qiime2-installation" class="section level3" number="5.0.2">
<h3>
<span class="header-section-number">5.0.2</span> QIIME2 installation<a class="anchor" aria-label="anchor" href="#qiime2-installation"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>We assumes that you have already installed QIIME 2. If not please do so using the instructions described <a href="https://docs.qiime2.org/2022.2/install/">here!</a>.</li>
</ul>
<p>A simple demo for installing qiime2 on Mac OS</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="qiime2-taxonomic-profiling.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://data.qiime2.org/distro/core/qiime2-2022.2-py38-osx-conda.yml</span>
<span id="cb20-2"><a href="qiime2-taxonomic-profiling.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> env create <span class="at">-n</span> qiime2-2022.2 <span class="at">--file</span> qiime2-2022.2-py38-osx-conda.yml</span>
<span id="cb20-3"><a href="qiime2-taxonomic-profiling.html#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> qiime2-2022.2-py39-osx-conda.yml</span>
<span id="cb20-4"><a href="qiime2-taxonomic-profiling.html#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="qiime2-taxonomic-profiling.html#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Activate qiime environment </span></span>
<span id="cb20-6"><a href="qiime2-taxonomic-profiling.html#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate qiime2-2022.2</span>
<span id="cb20-7"><a href="qiime2-taxonomic-profiling.html#cb20-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-8"><a href="qiime2-taxonomic-profiling.html#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Confirm installation</span></span>
<span id="cb20-9"><a href="qiime2-taxonomic-profiling.html#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> info</span></code></pre></div>
</div>
<div id="running-qiime2-commands" class="section level3" number="5.0.3">
<h3>
<span class="header-section-number">5.0.3</span> Running QIIME2 commands<a class="anchor" aria-label="anchor" href="#running-qiime2-commands"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>We prefer to use the command line interface <a href="https://docs.qiime2.org/2022.2/interfaces/q2cli/">g2cli</a>.</li>
<li>Activate the qiime2 environment: We are using qiime2-2022.2.</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="qiime2-taxonomic-profiling.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate qiime2-2022.2</span></code></pre></div>
</div>
<div id="import-paired-end-fastq-files" class="section level3" number="5.0.4">
<h3>
<span class="header-section-number">5.0.4</span> Import paired-end fastq files<a class="anchor" aria-label="anchor" href="#import-paired-end-fastq-files"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="qiime2-taxonomic-profiling.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools import <span class="dt">\</span></span>
<span id="cb22-2"><a href="qiime2-taxonomic-profiling.html#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--type</span> <span class="st">'SampleData[PairedEndSequencesWithQuality]'</span> <span class="dt">\</span></span>
<span id="cb22-3"><a href="qiime2-taxonomic-profiling.html#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--input-path</span> pe-33-manifest.tsv <span class="dt">\</span></span>
<span id="cb22-4"><a href="qiime2-taxonomic-profiling.html#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-path</span> qiime2_bushmeat/paired-end-demux.qza <span class="dt">\</span></span>
<span id="cb22-5"><a href="qiime2-taxonomic-profiling.html#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--input-format</span> PairedEndFastqManifestPhred33V2</span></code></pre></div>
<ul>
<li>Below is a minimal wrapper for
<ul>
<li>exploring the metadata,</li>
<li>importing the data,</li>
<li>sequence quality control,</li>
<li>sequence classification,</li>
<li>taxonomic assignment,</li>
<li>generating feature table.</li>
</ul>
</li>
</ul>
<p>For more information refer to <code>qiime2</code> <a href="https://docs.qiime2.org/2022.2/tutorials/">tutorials</a>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="qiime2-taxonomic-profiling.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/qiime</span></span>
<span id="cb23-2"><a href="qiime2-taxonomic-profiling.html#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="qiime2-taxonomic-profiling.html#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Usage: qiime2 code/qiime2_code.bash</span></span></code></pre></div>
</div>
<div id="qiime2-output-tidying" class="section level2" number="5.1">
<h2>
<span class="header-section-number">5.1</span> QIIME2 output tidying<a class="anchor" aria-label="anchor" href="#qiime2-output-tidying"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">taxlevels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"kingdom"</span>, <span class="st">"phylum"</span>, <span class="st">"class"</span>, <span class="st">"order"</span>, <span class="st">"family"</span>, <span class="st">"genus"</span>, <span class="st">"species"</span><span class="op">)</span>

<span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_delim</a></span><span class="op">(</span><span class="st">"/Volumes/RedSeagate/MOTHUR_BUSHMEAT/SRR7450/qiime2_bushmeat/q2-transformed-tables/taxonomy.tsv"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select_all.html">rename_all</a></span><span class="op">(</span><span class="va">tolower</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>otu <span class="op">=</span> <span class="st">"feature id"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>confidence <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">confidence</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="co"># filter(confidence == 1.00) %&gt;% </span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="st">"; s__$"</span>, <span class="st">""</span><span class="op">)</span>,
         taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="st">"; g__$"</span>, <span class="st">""</span><span class="op">)</span>,
         taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="st">"; f__$"</span>, <span class="st">""</span><span class="op">)</span>,
         taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="st">"; o__$"</span>, <span class="st">""</span><span class="op">)</span>,
         taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="st">"; c__$"</span>, <span class="st">""</span><span class="op">)</span>,
         taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="st">"; p__$"</span>, <span class="st">""</span><span class="op">)</span>,
         taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="st">"; k__$"</span>, <span class="st">""</span><span class="op">)</span>,
         <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html">separate</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">taxon</span>, into <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html">all_of</a></span><span class="op">(</span><span class="va">taxlevels</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"; "</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="st">"RDataRDS/q2_taxonomy.rds"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_delim</a></span><span class="op">(</span><span class="st">"/Volumes/RedSeagate/MOTHUR_BUSHMEAT/SRR7450/qiime2_bushmeat/q2-transformed-tables/feature-table.tsv"</span>, skip <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>otu <span class="op">=</span> <span class="st">"#OTU ID"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="st">"RDataRDS/q2_otutable.rds"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"RDataRDS/q2_taxonomy.rds"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"RDataRDS/q2_otutable.rds"</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"otu"</span><span class="op">)</span></code></pre></div>
<pre><code># A tibble: 494 × 19
   otu     kingdom phylum class order family genus species confidence SRR7450728
   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;        &lt;dbl&gt;      &lt;dbl&gt;
 1 04edaa… k__Bac… p__Fi… c__C… o__C… f__Pe… g__[… s__sor…       0.96        267
 2 e27b9e… k__Bac… p__Fi… c__B… o__B… f__St… g__M… s__cas…       0.91      15210
 3 d46e22… k__Bac… p__Pr… c__G… o__E… f__En… &lt;NA&gt;  &lt;NA&gt;          1          2374
 4 41db43… k__Bac… &lt;NA&gt;   &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;  &lt;NA&gt;          0.93        409
 5 9580aa… k__Bac… p__Fi… c__C… o__C… f__Cl… g__C… s__per…       1           146
 6 6f70e5… k__Bac… p__Fi… c__C… o__C… f__Cl… g__C… &lt;NA&gt;          0.8           0
 7 547357… k__Bac… &lt;NA&gt;   &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;  &lt;NA&gt;          0.97       7092
 8 bc3621… k__Bac… p__Pr… c__G… o__P… f__Mo… g__A… &lt;NA&gt;          1           357
 9 9209e4… k__Bac… p__Fi… c__C… o__C… f__Cl… g__C… &lt;NA&gt;          0.98        689
10 5970d8… k__Bac… p__Cy… c__C… o__C… &lt;NA&gt;   &lt;NA&gt;  &lt;NA&gt;          1           111
# … with 484 more rows, and 9 more variables: SRR7450732 &lt;dbl&gt;,
#   SRR7450746 &lt;dbl&gt;, SRR7450747 &lt;dbl&gt;, SRR7450752 &lt;dbl&gt;, SRR7450753 &lt;dbl&gt;,
#   SRR7450755 &lt;dbl&gt;, SRR7450757 &lt;dbl&gt;, SRR7450758 &lt;dbl&gt;, SRR7450759 &lt;dbl&gt;</code></pre>
<p><br><br></p>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="mothur-taxonomic-profiling.html"><span class="header-section-number">4</span> Mcrobial profiling Using mothur Pipeline</a></div>
<div class="next"><a href="metaphlan-taxonomic-profiling.html"><span class="header-section-number">6</span> Taxonomic Profiling of Metagenomics Sequencing Data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li>
<a class="nav-link" href="#qiime2-taxonomic-profiling"><span class="header-section-number">5</span> Mcrobial Profiling Using qiime2 Pipeline</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#download-a-qiime2-trained-classifer"><span class="header-section-number">5.0.1</span> Download a QIIME2 trained classifer</a></li>
<li><a class="nav-link" href="#qiime2-installation"><span class="header-section-number">5.0.2</span> QIIME2 installation</a></li>
<li><a class="nav-link" href="#running-qiime2-commands"><span class="header-section-number">5.0.3</span> Running QIIME2 commands</a></li>
<li><a class="nav-link" href="#import-paired-end-fastq-files"><span class="header-section-number">5.0.4</span> Import paired-end fastq files</a></li>
</ul>
</li>
<li><a class="nav-link" href="#qiime2-output-tidying"><span class="header-section-number">5.1</span> QIIME2 output tidying</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/CDI2021/smda_part2/blob/main/02b_bioinfomatics_analysis.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/CDI2021/smda_part2/edit/main/02b_bioinfomatics_analysis.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p><strong>Bioinformatics Analysis of Microbiome Data</strong> is PART 2 of the <a class="text-light" href="https://complexdatainsights.com/books/microbiome-analysis-in-R/"><u>End-to-End Practical User Guides</u></a> written by Teresia Mrema-Buza, A Microbiome Data Science Enthusiast and Owner of the Complex Data Insights, LLC, USA. 
  </p>
</div>

  <div class="col-12 col-md-6 mt-3">
    <p>This guide was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div>
</div>
</footer>
</body>
</html>
